auth/src/main/java/com/firebase/ui/auth/viewmodel/SignInViewModelBase.java

```diff
@@ -22,4 +25,19 @@ protected void setResult(Resource<IdpResponse> output) {
     protected void handleSuccess(@NonNull IdpResponse response, @NonNull AuthResult result) {
         setResult(Resource.forSuccess(response.withResult(result)));
     }
+
+    protected void handleMergeFailure(@NonNull AuthCredential credential) {
+        IdpResponse failureResponse = new IdpResponse.Builder(credential)
+                .build();
+        setResult(Resource.<IdpResponse>forFailure(new FirebaseAuthAnonymousUpgradeException(
```

Comment:
You could also make `FirebaseAuthAnonymousUpgradeException` have a constructor that takes `AuthCredential` since you're just converting the credential to an `IdpResponse`

Reply:
If not, this method should call the method below since they are the same `setResult` call

Reply:
Done.